testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,755 @@
+    }
+
+    /**
+     * Call this method to capture the state, save it, and do the diffing if all required
+     * states for the current test have been captured.
+     * @throws IOException 
+     */
+    public void runCapture() {
+        // if we already captured 4 states for the current test, then we are ready to diff
+        if (currentTestStates.size() == MAX_NUM) {
+            if (enableLogging) {
+                logger.startTimer();
+            }
+            if (enableLogging) {
+                logger.stopTimeAndUpdate(StateCaptureLogger.Task.DIFF);
+                logger.saveToFileAndReset();
+            }
+        }
+    }
+
+    /**
+     * Writes content into a file.
+     *
+     * @param  fn       name of the destination file
+     * @param  content  string representing the data to be written
+     * @param  append   boolean indicating whether to append to destination file or rewrite it
+     */
+    protected void writeToFile(String fn, String content, boolean append) {
+        synchronized(StateCapture.class) {
+            try {
+                File f = new File(fn);
+                f.createNewFile();
+
+                FileWriter fw = new FileWriter(f.getAbsoluteFile(), append);
+                BufferedWriter w = new BufferedWriter(fw);
+                w.write(content);
+                w.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    Set<String> File2SetString(String path) {
+        File file = new File(path);
+        Set<String> keys = new HashSet<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                keys.add(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return keys;
+    }
+
+    public void setup() {
+        subxmlFold = MainAgent.subxmlFold;
+        rootFold = MainAgent.rootFold;
+        diffFold = MainAgent.diffFold;
+        slug = MainAgent.slug;
+        reflectionFile = MainAgent.reflectionFold + "/0.txt";
+        xmlFileNum = countDirNums(subxmlFold);
+    }
+
+    public void fixing(String fieldName) throws IOException {
+        String subxml0 = subxmlFold + "/0xml";
+        try {
+            String path0 = subxml0 + "/" + fieldName + ".xml";
+            String state0 = readFile(path0);
+            String className = fieldName.substring(0, fieldName.lastIndexOf("."));
+            String subFieldName = fieldName.substring(fieldName.lastIndexOf(".")+1, fieldName.length());
+
+            Object ob_0;
+            XStream xstream = getXStreamInstance();
+
+            try {
+                Class c = Class.forName(className);
+                Field[] Flist = c.getDeclaredFields();
+                for (int i=0; i< Flist.length; i++) {
+                    if (Flist[i].getName().equals(subFieldName)) {
```

Comment:
Maybe change this to check the opposite (that it does not equal) and continue if so; doing so will ensure less indentation.